Generalization refactorings

guilhermeblanco opened the pull request:
Hi ServerGrove team,

My team have worked hard to turn this extension much more usable in a various ways.
Some enhancements are very good, but we had to drop ODM support for now since the team won't be using it, but we did our best to turn into very extendable and reusable for other mappings.
Here is a list of enhancements:
- Add Locale
- Drop Locale
- New entries now supports Domain and File
- ORM support
- XLIFF support

We hope you like it! =)

Regards,

Guilherme Blanco on behalf of InstaClick Inc.

guilhermeblanco committed:
Initial efforts around conversion from MongoDB specific to Storage agnostic (but implementing ORM as default).

guilhermeblanco committed:
Added missing field to Entry.

guilhermeblanco committed:
Refactored services initialization for consistency with Symfony standards.

shiroyuki committed:
Refactored the code. (This commit will break YML import.)

shiroyuki committed:
Laid out the code to support Xliff files with ORM.

guilhermeblanco committed:
Several fixes to entities and also initial efforts on controller.

shiroyuki committed:
Refactored the code ready for integrating with ORM.

guilhermeblanco committed:
Merge branch 'master' of github.com:instaclick/TranslationEditorBundle

guilhermeblanco committed:
More work around ImportCommand.

shiroyuki committed:
Added createLocale and updated the variable referencing AppKernel.

guilhermeblanco committed:
Finished import command. Still need to test if UTF-8 chars are imported correctly.

guilhermeblanco committed:
Refactored frontend. Most actions are working (except the new translation).

shiroyuki committed:
Refactored the code from ImportCommand to Base; finished the coding to analyze, classify the translation and rendered into a string in ExportCommand. Please note that ExportCommand in this commit cannot write a file.

shiroyuki committed:
Refactored the XLIFF file handler from the export command and make the file handler as a service.

guilhermeblanco committed:
Finished changes in Import and Export Commands. Now they are completely isolated.

guilhermeblanco committed:
Optimized ORMStorage and also added missing methods on StorageInterface. Also, moved ObjectManager compatible code to AbstractStorage.

guilhermeblanco committed:
Made new entry works.

guilhermeblanco committed:
Improved reusabiity of Importers. Added YamlImporter.

guilhermeblanco committed:
Committed missing file

guilhermeblanco committed:
Added YAML Exporter.

guilhermeblanco committed:
Added support to create and drop locales dynamically.

guilhermeblanco committed:
Turned domain into a drop down list.

guilhermeblanco committed:
{{@inheritdoc}} => {@inheritdoc} replacement. Thanks @asm89 for pointing out.

guilhermeblanco committed:
Fixed bug with localized exporting uner XLIFF.

guilhermeblanco committed:
Fixed original file name exporter in XLIFF exporter.

catacgc committed:
fix default locale parameter

catacgc committed:
update the docs

catacgc committed:
add the yaml exporter to the list of available services

catacgc committed:
fix yaml importer typo

catacgc committed:
add template checks for null values

catacgc committed:
refactor the importer to allow multi file type imports

delegate the import to an ImportInterface implementation that supports
the file type

catacgc committed:
add the yaml importer

asm89 commented:
Any news on this?

guilhermeblanco commented:
I have a pending patch that removes the necessity of importers configuration.
It also already update docs.
So as soon as possible (and the author updates PR with suggested changes), I'll merge and it will become ok to ServerGrove to merge.

catacgc committed:
move the file extension check in the abstract importer, change xlf to xliff

guilhermeblanco committed:
Merge pull request #1 from catacgc/master

Fix some configuration bugs and add some info about the configuration in README.md

guilhermeblanco commented:
Hi @pgodel !

I merged a new change @catacgc submitted.
Now it's much reach since you don't require to define the importer and also YAML exporter is running smoothly.

Any estimates when you're going to merge his?

guilhermeblanco committed:
Update Controller/EditorController.php

dinamic commented:
Last comment on this one has been a month ago. Any updates since then?

dinamic commented:
@pgodel ping

pgodel commented:
We have not had time to merge the PR yet, sorry! We have been swamped with work. If there is anybody willing to implement mongodb in the PR it would be great!

dinamic committed:
~ using text type for the translation

guilhermeblanco committed:
Merge pull request #2 from dinamic/master

Translation type varchar -> text

guilhermeblanco committed:
Updated composer.json.

dinamic committed:
~ display 'MISSING' if a translation does not exist or is empty

guilhermeblanco committed:
Merge pull request #3 from dinamic/master

bugfix: unable to set translation

cajoy committed:
Added the support for Symfony 2.1 by making the code automatically determine the extension based on the version of Symfony kernel.

guilhermeblanco committed:
Update master

shiroyuki committed:
Change the interface to allow "getFileExtension" to return either a string or an array.

shiroyuki committed:
Added the support for legacy extensions (e.g., XLIFF and XLF extensions)

Commit:
added debug code

shiroyuki committed:
Added support for Symfony 2.1 and 2.0 (BC).

Commit:
removed the debugging code, properly support 2.0 and 2.1 and took out the array support.

shiroyuki committed:
Added the missing module.

guilhermeblanco committed:
Removed dependency of ORM EntityRepository, allowing custom implementations of EntityRepository to not require a non-interfaced Common Persistence method createQueryBuilder.

gimler commented:
ping. warm up this topic again.

gimler committed:
replace Translation implementation with symfony translator service

gimler committed:
update code for bootstrap 2

gimler committed:
remove inline js

guilhermeblanco committed:
Merge pull request #5 from gimler/bootstrap2

Bootstrap2

guilhermeblanco committed:
Merge pull request #4 from gimler/translator

Use symfony translation service

gimler commented:
this ticket fix #5 and #1

Commit:
Fixing addTranslationAction()

ch3ric committed:
fixed typo

robocoder committed:
Merge pull request #6 from ch3ric/patch-1

fixed typo

robocoder committed:
update composer.json

ch3ric committed:
removed useless conf (exporter)

ch3ric committed:
removed useless conf (exporter)

ch3ric committed:
removed useless conf (exporter)

ch3ric committed:
The catalogue domain corresponds to the fileName

guilhermeblanco committed:
Merge pull request #7 from ch3ric/master

Removed useless conf and fixed filename during export

damienalexandre commented:
Great PR but:
- Documentation is not up-to-date, refer to Mongo, Orm, but do not expose the right one;
- Why the f**\* Pablo's name is removed from the composer file? `authors` accept an array;
- Classes declared as services in `Resources/config/services.xml` should be parameters.

Beside that, ping @servergrove - this definitly deserve a merge :+1:

pgodel commented:
We are still swamped with work. Hopefully will be able to dedicate some time soon, sorry.

ynx committed:
Update ImportCommand.php

ynx committed:
Update AbstractStorage.php

ynx committed:
Update StorageInterface.php

robocoder committed:
Merge pull request #8 from yxie/master

This patch will reduce the TranslationImport command's running time from 45+ mins to 1.5 mins.

robocoder committed:
Update composer.json

Commit:
1) Fixing issue that entity aren't persisted; 2) further optimize performance;

Commit:
3) Fixing the JavaScript that wrong locale is deleted by the delete buttons

guilhermeblanco committed:
Merge pull request #9 from yxie/master

Fixing various bugs during use of the bundle

michaelperrin commented:
:+1: Nice PR

alex88 committed:
Updated bootstrap and jquery assets url

Bootstrap urls were giving 404 and I've updated them with netdna hosted version and also the new jquery version

robocoder committed:
Merge pull request #10 from alex88/patch-1

Updated bootstrap and jquery assets url

Note: the reason the bootstrap URLs are 404'ing is due to https://github.com/blog/1452-new-github-pages-domain-github-io

gimler commented:
ping

servergrove commented:
Hi Gordon.

We can't merge this PR without mongodb support and we haven't been able to implement it yet. Not sure when we will have time. If you or someone can do it, it will be very much welcomed. 

Pablo Godel

On Apr 17, 2013, at 9:29, Gordon Franke notifications@github.com wrote:

> ping
> 
> —
> Reply to this email directly or view it on GitHub.

Commit:
Fixed filer to accept anything before the first dot

guilhermeblanco committed:
Merge pull request #12 from ETSGlobal/fix-file-finder

Fixed filer to accept anything before the first dot

guilhermeblanco committed:
Update composer.json

pabgaran committed:
Update ImportCommand.php

Add support to import files like messages.es_ES.yml

pabgaran committed:
Avoid error on empty value

guilhermeblanco committed:
Merge pull request #13 from pgarrido/patch-1

Update ImportCommand.php

pabgaran committed:
Update template to bootstrap 3.0 Step 1

pabgaran committed:
Update template to bootstrap 3.0 and add search capability

pabgaran committed:
Fix function names

guilhermeblanco committed:
Update README.md

Modifications for #14

guilhermeblanco committed:
Merge pull request #14 from pgarrido/master

Update to bootstrap 3 and add search capability

ivch committed:
saveLocal & saveEntry fix

In current version there is a problem with "Create Locale/Entry" forms.
Actually ajax request for saving doesn't works because saveEntryForm() and saveLocaleForm() receive wrong object

guilhermeblanco committed:
Merge pull request #15 from ivch/patch-1

saveLocal & saveEntry fix

pasinter committed:
MonogDB storage support

guilhermeblanco committed:
Merge pull request #16 from pasinter/master

Added MonogDB storage support
